enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/InteractiveQuestion.NotebookButton.unit.spec.tsx

```diff
@@ -0,0 +1,110 @@
+import userEvent from "@testing-library/user-event";
+
+import {
+  setupAlertsEndpoints,
+  setupCardEndpoints,
+  setupCardQueryMetadataEndpoint,
+  setupCardQueryEndpoints,
+  setupDatabaseEndpoints,
+  setupTableEndpoints,
+} from "__support__/server-mocks";
+import { renderWithProviders, screen } from "__support__/ui";
+import { createMockJwtConfig } from "embedding-sdk/test/mocks/config";
+import { setupSdkState } from "embedding-sdk/test/server-mocks/sdk-init";
+import {
+  createMockCard,
+  createMockCardQueryMetadata,
+  createMockColumn,
+  createMockDatabase,
+  createMockDataset,
+  createMockDatasetData,
+  createMockTable,
+  createMockUser,
+} from "metabase-types/api/mocks";
+
+import { InteractiveQuestion } from "./InteractiveQuestion";
+
+const TEST_USER = createMockUser();
+const TEST_DB_ID = 1;
+const TEST_DB = createMockDatabase({ id: TEST_DB_ID });
+
+const TEST_TABLE_ID = 1;
+const TEST_TABLE = createMockTable({ id: TEST_TABLE_ID, db_id: TEST_DB_ID });
+
+const TEST_COLUMN = createMockColumn({
+  display_name: "Test Column",
+  name: "Test Column",
+});
+
+const TEST_DATASET = createMockDataset({
+  data: createMockDatasetData({
+    cols: [TEST_COLUMN],
+    rows: [["Test Row"]],
+  }),
+});
+
+const setup = ({
+  isOpen = true,
+}: {
+  isOpen?: boolean;
+} = {}) => {
+  const { state } = setupSdkState({
+    currentUser: TEST_USER,
+  });
+
+  const TEST_CARD = createMockCard({
+    can_delete: true,
+    enable_embedding: true,
+  });
+
+  setupCardEndpoints(TEST_CARD);
+  setupCardQueryMetadataEndpoint(
+    TEST_CARD,
+    createMockCardQueryMetadata({
+      databases: [TEST_DB],
+      tables: [TEST_TABLE], // to be editable, card must have table and databse metadata
+    }),
+  );
+
+  setupAlertsEndpoints(TEST_CARD, []);
+  setupDatabaseEndpoints(TEST_DB);
+
+  setupTableEndpoints(TEST_TABLE);
+
+  setupCardQueryEndpoints(TEST_CARD, TEST_DATASET);
+
+  const clickSpy = jest.fn();
+
+  renderWithProviders(
+    <InteractiveQuestion questionId={TEST_CARD.id}>
+      <div>Look! A Button! 👇</div>
+      <InteractiveQuestion.NotebookButton onClick={clickSpy} isOpen={isOpen} />
+    </InteractiveQuestion>,
+    {
+      mode: "sdk",
+      sdkProviderProps: {
+        config: createMockJwtConfig({
+          jwtProviderUri: "http://TEST_URI/sso/metabase",
+        }),
+      },
+      storeInitialState: state,
+    },
+  );
+
+  return { clickSpy };
+};
+
+describe("InteractiveQuestion.NotebookButton", () => {
+  it("should render the notebook button", async () => {
+    setup({ isOpen: true });
+
+    expect(await screen.findByLabelText("notebook icon")).toBeInTheDocument();
```

Comment:
It's essential for this test that `QuestionNotebookButton.shouldRender` runs somewhere under the hood.
We can future-proof this test a bit by spying on `QuestionNotebookButton.shouldRender` and making sure it gets called at least once.

Reply:
👍  [937c3d7](https://github.com/metabase/metabase/pull/46350/commits/937c3d7e2a6d289ac527305534a3106b55917105)